Content.Server/GameObjects/Components/Conveyor/ConveyorComponent.cs

```diff
@@ -0,0 +1,288 @@
+            }
+
+            return true;
+        }
+
+        private bool CanMove(IEntity entity)
+        {
+            if (entity == Owner)
+            {
+                return false;
+            }
+
+            if (!entity.TryGetComponent(out ICollidableComponent collidable) ||
+                collidable.Anchored)
+            {
+                return false;
+            }
+
+            if (entity.HasComponent<ConveyorComponent>())
+            {
+                return false;
+            }
+
+            if (entity.HasComponent<IMapGridComponent>())
+            {
+                return false;
+            }
+
+            if (ContainerHelpers.IsInContainer(entity))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public void Update(float frameTime)
+        {
+            if (!CanRun())
+            {
+                return;
+            }
+
+            var intersecting = _entityManager.GetEntitiesIntersecting(Owner, true);
+            var direction = GetAngle().ToVec();
+            var speed = _speed * frameTime;
+
+            foreach (var entity in intersecting)
+            {
+                if (!CanMove(entity))
+                {
+                    continue;
+                }
+
+                if (entity.TryGetComponent(out ICollidableComponent collidable))
+                {
+                    var controller = collidable.EnsureController<ConveyedController>();
+                    controller.Move(direction, speed);
+                }
+            }
+        }
+
+        private bool ToolUsed(IEntity user, ToolComponent tool)
+        {
+            if (!Owner.HasComponent<ItemComponent>() &&
+                tool.UseTool(user, Owner, ToolQuality.Prying))
+            {
+                State = ConveyorState.Loose;
+
+                Owner.AddComponent<ItemComponent>();
+                Group?.RemoveConveyor(this);
+                Owner.Transform.WorldPosition += (_random.NextFloat() * 0.4f - 0.2f, _random.NextFloat() * 0.4f - 0.2f);
+
+                return true;
+            }
+
+            return false;
+        }
+
+        public void SyncState(ConveyorState state)
+        {
+            if (State == ConveyorState.Loose)
+            {
+                return;
+            }
+
+            State = state == ConveyorState.Loose
+                ? ConveyorState.Off
+                : state;
+        }
+
+        private void DeItemize(GridCoordinates coordinates)
+        {
+            if (!Owner.HasComponent<ItemComponent>())
+            {
+                return;
+            }
+
+            Owner.Transform.GridPosition = coordinates;
+            Owner.RemoveComponent<ItemComponent>();
+            State = ConveyorState.Off;
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(this, conveyor => conveyor.Id, "id", null);
```

Comment:
Getting IDs from YAML like this means that the ID loaded may conflict with IDs that are already used in the current round or such.

Reply:
What would be a better way to get the id from a saved map?

Reply:
Isn't really an easy way to short of storing the entity UID of the linked conveyor.